web/satellite/src/components/project/ProjectDetails.vue

```diff
@@ -0,0 +1,285 @@
+    import { Component, Vue } from 'vue-property-decorator';
+    import Button from '@/components/common/Button.vue';
+    import HeaderedInput from '@/components/common/HeaderedInput.vue';
+    import Checkbox from '@/components/common/Checkbox.vue';
+    import EmptyState from '@/components/common/EmptyStateArea.vue';
+    import { PROJETS_ACTIONS, APP_STATE_ACTIONS, NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
+    import ROUTES from '@/utils/constants/routerConstants';
+    import DeleteProjectPopup from '@/components/project/DeleteProjectPopup.vue';
+    import ProjectNavigation from '@/components/project/ProjectNavigation.vue';
+
+    @Component(
+        {
+            data: function () {
+                return {
+                    isEditing: false,
+                    newDescription: '',
+                };
+            },
+            methods: {
+                toggleEditing: function (): void {
+                    this.$data.isEditing = !this.$data.isEditing;
+                    // TODO: cache this value in future
+                    this.$data.newDescription = '';
+                },
+                setNewDescription: function (value: string): void {
+                    this.$data.newDescription = value;
+                },
+                onSaveButtonClick: async function (): Promise<any> {
+                    let response = await this.$store.dispatch(
+                        PROJETS_ACTIONS.UPDATE, {
+                            id: this.$store.getters.selectedProject.id,
+                            description: this.$data.newDescription,
+                        }
+                    );
+
+                    response.isSuccess
+                        // TODO: call toggleEditing method instead of this IIF
+                        ? (() => {
+                            this.$data.isEditing = !this.$data.isEditing;
+                            this.$data.newDescription = '';
+                            this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Project updated successfully!');
+                        })()
+                        : this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, response.errorMessage);
+                },
+                toggleDeleteDialog: function (): void {
+                    this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_DEL_PROJ);
+                },
+                onMoreClick: function (): void {
+                    this.$router.push(ROUTES.USAGE_REPORT);
+                },
+            },
+            computed: {
+                name: function (): string {
+                    return this.$store.getters.selectedProject.name;
+                },
+                description: function (): string {
+                    return this.$store.getters.selectedProject.description ?
+                        this.$store.getters.selectedProject.description :
+                        'No description yet. Please enter some information about the project if any.';
```

Comment:
const defaultDescription = 'No description yet. Please enter some information about the project if any.'
                    return this.$store.getters.selectedProject.description ?
                        this.$store.getters.selectedProject.description :
                        defaultDescription